pkg/controller/ybcluster/ybcluster_controller.go

```diff
@@ -683,6 +684,10 @@ func addDefaults(spec *yugabytev1alpha1.YBClusterSpec) {
 		}
 	}
 
+	if &spec.Domain == nil {
```

Comment:
```suggestion
	if &spec.Domain == nil || len(spec.Domain) == 0 {
```

When spec.Domain is missing, Go sets its value to blank string i.e. zero value of string type. So, &spec.Domain is not nil in that case, hence we need to check for the value as well.